client/components/date-range/index.js

```diff
@@ -0,0 +1,300 @@
+		) {
+			this.onSelectDate( date );
+		}
+	}
+
+	onSelectDate( date ) {
+		// DateUtils requires a range object with this shape
+		const range = {
+			from: this.momentDateToNative( this.state.startDate ),
+			to: this.momentDateToNative( this.state.endDate ),
+		};
+
+		const rawDay = this.momentDateToNative( date );
+
+		// Calculate the new Date range
+		const newRange = DateUtils.addDayToRange( rawDay, range );
+
+		// Edge case: Range can sometimes be from: null, to: null
+		if ( ! newRange.from || ! newRange.to ) return;
+
+		this.setState(
+			previousState => {
+				let newState = {
+					startDate: this.nativeDateToMoment( newRange.from ),
+					endDate: this.nativeDateToMoment( newRange.to ),
+					inputFromDate: this.nativeDateToMoment( newRange.from ).format( 'L' ),
+					inputToDate: this.nativeDateToMoment( newRange.to ).format( 'L' ),
+				};
+
+				// For first date selection only: "cache" previous dates
+				// just in case user doesn't "Apply" and we need to revert
+				// to the original dates
+				if ( ! this.state.oldDatesSaved ) {
+					newState = Object.assign( {}, newState, {
+						oldStartDate: previousState.startDate,
+						oldEndDate: previousState.endDate,
+						oldDatesSaved: true, // marks that we have saved old dates
+					} );
+				}
+
+				return newState;
+			},
+			() => this.props.onDateSelect( this.state.startDate, this.state.endDate )
+		);
+	}
+
+	revertDates() {
+		this.setState( previousState => {
+			const newState = { oldDatesSaved: false };
+
+			if ( previousState.oldStartDate && previousState.oldEndDate ) {
+				newState.startDate = previousState.oldStartDate;
+				newState.endDate = previousState.oldEndDate;
+			}
+
+			return newState;
+		} );
+	}
+
+	commitDates() {
+		this.setState(
+			previousState => ( {
+				oldStartDate: previousState.startDate, // update cached old dates
+				oldEndDate: previousState.endDate, // update cached old dates
+				oldDatesSaved: false,
+			} ),
+			() => {
+				this.props.onDateCommit( this.state.startDate, this.state.endDate );
+				this.togglePopover();
+			}
+		);
+	}
+
+	momentDateToNative( momentDate ) {
+		return momentDate.toDate();
+	}
+
+	nativeDateToMoment( nativeDate ) {
+		return this.props.moment( nativeDate );
+	}
+
+	dateToHumanReadable( date ) {
+		return this.props.moment( date ).format( 'L' );
+	}
+
+	renderDatePicker() {
+		const now = new Date();
+
+		return (
+			<DatePicker
+				className="date-range__popover-date-picker"
+				showOutsideDays={ false }
+				toMonth={ now }
+				onSelectDay={ this.onSelectDate }
+				selectedDays={ {
+					from: this.momentDateToNative( this.state.startDate ),
+					to: this.momentDateToNative( this.state.endDate ),
+				} }
+				numberOfMonths={ window.matchMedia( '(min-width: 480px)' ).matches ? 2 : 1 }
+				calendarViewDate={ this.momentDateToNative( this.state.startDate ) }
+				disabledDays={ [
+					{
+						after: now, // you can't look at photos from the future!
```

Comment:
This no longer really applies right? As a general component we should allow for future dates too and consider how best to limit days via props... We could allow `disabledDays` to be passed as props or have set restrictions via props like `allowFutureDates` & `allowPastDates` or similar.

Reply:
I think `disabledDays` prop.

Reply:
@spen Please see https://github.com/Automattic/wp-calypso/pull/29938/commits/142cc8b2ef58e994bea9b0d3c2fcd0aaf81619d9